src/validations/html/runRules.js

```diff
@@ -0,0 +1,40 @@
+import {SAXParser} from 'parse5';
+import voidElements from 'void-elements';
+
+// Runs `rules` on `source` and promises an iterable of all errors.
+//
+// `rules` must be an array of objects that satisfy the interface:
+// * done() returns an iterable of errors
+// * optionally,
+//   openTag({row: number, column: number}, {name: string})
+//   to take in open tags that are neither void nor self-closing
+// * optionally,
+//   closeTag({row: number, column: number}, name: string)
+//   to take in close tags
+export default (rules, source) => {
+  const parser = new SAXParser({locationInfo: true});
+  return new Promise((resolve) => {
+    parser.on('startTag', (name, attrs, selfClosing, {line, col}) => {
```

Comment:
That said, putting e.g. `<div />` in the branch right now does correctly generate an error saying that doesn’t work, so maybe 🤷‍♂️